src/spring-cloud/azext_spring_cloud/_validators.py

```diff
@@ -139,3 +143,99 @@ def validate_log_since(namespace):
+    _validate_cidr_range(namespace)
+
+    if namespace.vnet:
+        vnet = namespace.vnet
+        # format the app_subnet and service_runtime_subnet
+        if not is_valid_resource_id(vnet):
+            if vnet.count('/') > 0:
+                raise CLIError('vnet {0} is not a valid name or resource ID'.format(vnet))
+            vnet = resource_id(
+                subscription=get_subscription_id(cmd.cli_ctx),
+                resource_group=namespace.resource_group,
+                namespace='Microsoft.Network',
+                type='virtualNetworks',
+                name=vnet
+            )
+        namespace.app_subnet = _parse_subnet(vnet, namespace.app_subnet)
+        namespace.service_runtime_subnet = _parse_subnet(vnet, namespace.service_runtime_subnet)
+    else:
+        app_vnet_id = _validate_subnet_id(namespace.app_subnet)
+        service_runtime_vnet_id = _validate_subnet_id(namespace.service_runtime_subnet)
+        if app_vnet_id.lower() != service_runtime_vnet_id.lower():
+            raise CLIError('--app-subnet and --service-runtime-subnet should be in the same Virtual Networks.')
+    if namespace.app_subnet.lower() == namespace.service_runtime_subnet.lower():
+        raise CLIError('--app-subnet and --service-runtime-subnet should not be the same.')
+
+
+def _validate_subnet_id(subnet_id):
+    if not is_valid_resource_id(subnet_id):
+        raise CLIError('subnet {0} is not a valid resource ID'.format(subnet_id))
+    subnet = parse_resource_id(subnet_id)
+    return resource_id(
+        subscription=subnet['subscription'],
+        resource_group=subnet['resource_group'],
+        namespace=subnet['namespace'],
+        type=subnet['type'],
+        name=subnet['name']
+    )
+
+
+def _parse_subnet(vnet_id, subnet):
+    if not is_valid_resource_id(subnet):
+        if subnet.count('/'):
+            raise CLIError('subnet {0} is not a valid name or resource ID'.format(subnet))
+        # subnet name is given
+        return vnet_id + '/subnets/' + subnet
+    if not subnet.lower().startswith(vnet_id.lower()):
+        raise CLIError('subnet {0} is not under virtual network {1}'.format(subnet, vnet_id))
+    return subnet
+
+
+def _validate_cidr_range(namespace):
+    ranges = namespace.reserved_cidr_range.split(',')
+    ranges = [x for x in ranges if x != '']  # filter out empty ones
+
+    if len(ranges) == 1:
+        _validate_ip(ranges[0], 14)
+        namespace.reserved_cidr_range = ranges[0]
+        return
+    if len(ranges) != 3:
+        raise CLIError('--reserved-cidr-range should be 1 unused /14 IP range, or 3 unused /16 IP rangeds')
+    ipv4 = [_validate_ip(ip, 16) for ip in ranges]
+    # check no overlap with each other
+    for i, item in enumerate(ipv4):
+        for j in range(i + 1, len(ipv4)):
+            if item.overlaps(ipv4[j]):
+                raise CLIError('--reserved-cidr-range should not be overlapped with each other. But got {0} and {1}'

```

Comment:
'--reserved-cidr-range should not overlap each other, but {0} and {1} overlapping'